### DISCUSSION TITLE
Typescript code with option noImplicitAny:true

### DISCUSSION THREAD

**Discussion by rla-dev** [avajs/ava#3197] [Q&A] [answered]
I compile my Typescript tests with the compiler option noImplicitAny:true. Using tsc 4.9.5 (or 5.0.4) I get the following error:
 ```Parameter 't' implicitly has an 'any' type. ```

Actually I would  sometimes like to type the parameter t as follows:
```
  (t: ExecutionContext<MyContext>) => { ... }
```
I haven't found out to import AVAs ExecutionContext (import {ExecutionContext} from "ava" does work work).
What did I miss?

**Comment by novemberborn** [comment, marked as answer]
> I compile my Typescript tests with the compiler option noImplicitAny:true. Using tsc 4.9.5 (or 5.0.4) I get the following error:
> `Parameter 't' implicitly has an 'any' type. `

Would be good to see the code pattern that triggers that. `t` is typed as `ExecutionContext<Context = unknown>` which isn't `any`.



> Actually I would sometimes like to type the parameter t as follows:
> 
> ```
>   (t: ExecutionContext<MyContext>) => { ... }
> ```
> 
> I haven't found out to import AVAs ExecutionContext (import {ExecutionContext} from "ava" does work work). What did I miss?

https://github.com/avajs/ava/blob/main/docs/recipes/typescript.md#typing-tcontext

**Reply by rla-dev** [reply]
@novemberborn Thanks for pointing to the example in the docs. I was expecting a different way for parameterization the context, but following the example solved my issue.
